lib/bolt/config.rb

```diff
@@ -154,12 +181,39 @@ def initialize(boltdir, config_data, overrides = {})
         @transports[key] = transport.default_options
       end
 
+      config_data = merge_config_data(config_data)
+
       update_from_file(config_data)
       apply_overrides(overrides)
 
       validate
     end
 
+    # Merge configuration
+    # Precedence from highest to lowest is: project, user-level, system-wide
+    def merge_config_data(config_data)
+      config_data.each_with_object({}) do |config, acc|
+        @config_files << config[:filepath] if config[:filepath]
+
+        acc.merge!(config[:data]) do |key, val1, val2|
```

Comment:
We should change these `merge!` calls to just `merge`. That requires getting rid of `each_with_object`, but that's straightforward if the `@config_files << [...]` is removed from this method.